app/models/tenant.rb

```diff
@@ -220,6 +220,10 @@ def self.root_tenant
     in_my_region.roots.first
   end
 
+  def self.root_tenant_name
+    @root_tenant_name ||= root_tenant.name
```

Comment:
Ah, I see specs fail horribly after memoizing `root_tenant`.

What about if we do the same magic as the `MiqServer.myserver` does:

```
diff --git a/app/models/tenant.rb b/app/models/tenant.rb
index 5c53243..7d1ef32 100644
--- a/app/models/tenant.rb
+++ b/app/models/tenant.rb
@@ -216,13 +216,11 @@ class Tenant < ApplicationRecord
   # from this tenant, all tenants are positioned
   #
   # @return [Tenant] the root tenant
-  def self.root_tenant
-    in_my_region.roots.first
-  end
+  cache_with_timeout(:root_tenant) { in_my_region.roots.first }
 
   # NOTE: returns the root tenant
   def self.seed
-    root_tenant || create!(:use_config_for_attributes => true) do |_|
+    in_my_region.roots.first || create!(:use_config_for_attributes => true) do |_|
       _log.info("Creating root tenant")
     end
   end
diff --git a/spec/support/evm_spec_helper.rb b/spec/support/evm_spec_helper.rb
index 2eafc0c..9ccb6a3 100644
--- a/spec/support/evm_spec_helper.rb
+++ b/spec/support/evm_spec_helper.rb
@@ -77,6 +77,7 @@ module EvmSpecHelper
   end
 
   def self.remote_miq_server(attrs = {})
+    Tenant.root_tenant_clear_cache
     Tenant.root_tenant || Tenant.create!(:use_config_for_attributes => false)
 
     FactoryGirl.create(:miq_server, attrs)
```